chain/store/stateGenerator.go

```diff
@@ -266,6 +290,28 @@ func (cs *ChainStore) generateStateRoot(ctx context.Context, b *block.Block, gen
 				if err = states.UpdateID(pg[0], id); err != nil {
 					return nil, common.EmptyUint256, err
 				}
+
+			case pb.PayloadType_GENERATE_ID_2_TYPE:
+				select {
+				case <-ctx.Done():
+					return nil, common.EmptyUint256, errors.New("context deadline exceeded")
```

Comment:
You can change `errors.New("context deadline exceeded")` to `ctx.Error()`, same for id1 case